ultravox/tools/infer_tool.py

```diff
@@ -190,25 +192,48 @@ def dataset_infer(inference: base.VoiceInference, args: InferArgs):
     if args.seed is not None:
         ds_args.shuffle_seed = args.seed
     ds = datasets.create_dataset(args.data_sets[0], ds_args)
-    scores: List[float] = []
-    for i, sample in enumerate(datasets.Range(ds, args.num_samples)):
-        # Store the original question and answer for JSON output.
-        question_text = sample.audio_transcript
-        expected_answer = sample.messages[-1]["content"]
-        # Drop any assistant response from the sample.
-        sample.messages = sample.messages[:-1]
-        if not args.json:
-            run_tui(i, inference, sample, args, expected_answer, scores)
-        else:
-            output = inference.infer(
-                sample, max_tokens=args.max_tokens, temperature=args.temperature
+
+    if args.json and isinstance(inference, infer.LocalInference):
+        start_time = time.time()
+        dl = DataLoader(
+            datasets.Range(ds, args.num_samples),
+            batch_size=args.batch_size,
+            collate_fn=lambda x: x,
+        )
+        current_batch = []
+        sample_index = 0
+        for current_batch in dl:
+            output = []
+            for sample in current_batch:
+                output.append(
+                    {
+                        "index": sample_index,
+                        "question_text": sample.audio_transcript,
+                        "expected_answer": sample.messages[-1]["content"],
+                    }
+                )
+                sample.messages = sample.messages[:-1]
+                sample_index += 1
+
+            output_batch = inference.batch_infer(
+                current_batch,
+                max_tokens=args.max_tokens,
+                temperature=args.temperature,
             )
-            obj = {
-                "question": question_text,
-                "generated_answer": output.text,
-                "expected_answer": expected_answer,
-            }
-            print(json.dumps(obj))
+            for i, output_text in enumerate(output_batch):
+                output[i]["output_text"] = output_text
+            print(output)
+        print("Total time", time.time() - start_time)
```

Comment:
I don't think we want to mix JSON and freeform text.

Reply:
Why is it bad to mix the two? Aren't they just logs at the end of the day?

Reply:
For my own understanding, why is it bad to mix JSON and freeform text? Aren't they just logs at the end of the day?

Reply:
a JSONL parser will explode if it gets non-JSON text, right?

Reply:
Is it common to use a jsonl parser for logs? I thought logs are mostly for debugging purpose, and not meant for further post processing